File: backend/src/main/java/com/app/mevocab/internal/user/dto/response/UserResponse.java
package com.app.mevocab.internal.user.dto.response;

import com.app.mevocab.common.enums.UserRole;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserResponse {
    String userID;
    String username;
    String name;
    String email;
    UserRole role;
}

File: frontend/lib/data/models/dto/UserResponse.dart
class UserResponse {
  final String userID;
  final String username;
  final String name;
  final String email;
  final String role;

  UserResponse({
    required this.userID,
    required this.username,
    required this.name,
    required this.email,
    required this.role,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      userID: json['userID'] ?? '',
      username: json['username'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      role: json['role'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userID': userID,
      'username': username,
      'name': name,
      'email': email,
      'role': role,
    };
  }
}
